ui/src/router.js

```diff
@@ -118,20 +141,88 @@ router.beforeEach((to, from, next) => {
     const config = require("@/config/config.json");
 
     if (
-      to.path == "/" ||
+      
       to.path == "/login" ||
       to.path == "/logout" ||
       config.demo
     ) {
       next();
-    } else {
+    } 
+    else {
       next({
         path: "/login"
       });
     }
   } else {
-    next();
+    if(
```

Comment:
I don't think you need this conditional. You can just have checkPageRole return true if we don't have a matching path for it.